docs/iot/tutorials/gpio-input.md

```diff
@@ -0,0 +1,142 @@
+
+:::image type="content" source="https://via.placeholder.com/600x400.png" alt-text="Animated GIF showing magnetic reed switch":::
+
+## Laser tripwire
+
+Extending the previous example concept a bit further, let's take a look at how this could be applied to creating a laser tripwire. Building a laser tripwire requires the following additional components:
+
+* KY-008 laser transmitter module
+* Laser receiver sensor module *(see note below)*
+* 2 10K Ω resistors
+
+> [!NOTE]
+> *Laser receiver sensor module* is the generic name applied to a common module found at many internet retailers. The device may vary in name or manufacturer, but should resemble this image.
+>
+> ![Image of a Laser Receiver Sensor Module](https://via.placeholder.com/200x200.png)
+
+### Connect laser tripwire hardware
+
+Connect the components as detailed in following diagram.
+
+:::image type="content" source="https://via.placeholder.com/500x300.png" alt-text="Fritzing diagram":::
+
+Pay close attention to the 10K Ω resistors. These implement a [voltage divider](https://www.seeedstudio.com/blog/2019/10/09/voltage-dividers-everything-you-need-to-know/). This is because the laser receiver module outputs 5V to indicate the beam is broken. Raspberry Pi only supports up to 3.3V for GPIO input. Since sending the full 5V to the pin could damage the Raspberry Pi, the current from the receiver module is passed through a voltage divider to halve the voltage to 2.5V.
+
+### Apply source code updates
+
+You can *almost* use the same code as earlier, with one exception. In the other examples, we used `PinMode.InputPullUp` so that when the pin is disconnected from ground and the circuit is open, the pin returns `PinValue.High`.
+
+However, in the case of the laser receiver module, we're not detecting an open circuit. Instead, we want the pin to act as a sink for current coming from the laser receiver module. In this case, we'll open the pin with `PinMode.InputPullDown`. This way, the pin returns `PinValue.Low` when it's getting no current, and `PinValue.High` when it receives current from the laser receiver module.
+
+```csharp
+controller.OpenPin(pin, PinMode.InputPullDown);
+```
+
+> [!IMPORTANT]
+> Make sure the code deployed on your Raspberry Pi includes this change before testing a laser tripwire. The program *does* work without it, but using the wrong input mode risks damage to your Raspberry Pi!
+
+:::image type="content" source="https://via.placeholder.com/600x400.png" alt-text="Animated GIF showing the laser tripwire":::
+
+## Get the source code
+
+The source for this tutorial is [available on GitHub](https://github.com/MicrosoftDocs/dotnet-iot-assets/tree/main/tutorials/InputTutorial).
```

Comment:
This area is where I'd give readers some ideas for how to apply it. Detecting a switch can do a lot of different things. Just some brainstorming:

- turning on indicators / displays.
- outside lighting / auto dusk-dawn turn on / turn off
- do not disturb
- bring coffee...

The laser is one of those ideas.